2_generics_collections/live_coding/bank_1/src/bank.rs

```diff
@@ -0,0 +1,67 @@
+use std::collections::HashMap;
+
+pub type Money = u64;
+
+pub struct Bank {
+    accounts: HashMap<String, Money>,
+}
+
+impl Bank {
+    pub fn new() -> Bank {
+        Bank {
+            accounts: HashMap::new(),
+        }
+    }
+
+    /// Return true if the account was created successfully, false otherwise
+    pub fn add_account(&mut self, account_name: &str) -> bool {
+        match self.accounts.get(account_name) {
+            None => {
+                self.accounts.insert(String::from(account_name), 0);
+                true
+            }
+            Some(_) => false,
+        }
+    }
+
+    pub fn add_money(&mut self, account_name: &str, money_amount: Money) {
+        let current_amount = match self.accounts.get(account_name) {
+            None => 0,
+            Some(amount) => *amount,
+        };
+
+        self.accounts.insert(
+            String::from(account_name),
+            current_amount.clone() + money_amount,
+        );
+    }
```

Comment:
From what I understand, `accounts.insert()` creates an account. I don't think we should create an account if it did not already existed. Maybe using [get_mut](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.get_mut) is a better idea ? We should return an error (bool ?) if the operation did not succeed.